tests/test_sparse_tensors.py

```diff
@@ -209,7 +209,10 @@ def test_sparse_softmax(tensor_type, device):
     res_sparse.values().backward(torch.ones_like(res_sparse.values()))
     res_gt.backward(torch.ones_like(res_gt))
 
-    assert torch.allclose(a.grad, a_sparse.grad.to_dense(), atol=2e-7)
+    a_grad = a.grad.clone()
+    a_grad[~mask] = 0
```

Comment:
This was a "bug" in the test before, as PyTorch implementation of `softmax` yields `inf` / `nan` in the backward if a full row is empty.

This makes me think that we should probably provide a "safe-softmax` alternative in xformers. It won't be very efficient if done with pure pytorch operators, but I've already found myself in need of such a function and thus it would be handy to have it somewhere